docs/deploy/nginx-uwsgi.rst

```diff
@@ -0,0 +1,219 @@
+dependencies.
+
+.. code:: sh
+
+  $ python3 -m venv /home/myproject/venv
+
+Then install your dependencies.
+
+.. code:: sh
+
+  $ /home/myproject/venv/bin/pip install -r /home/myproject/src/requirements.txt
+  $ /home/myproject/venv/bin/pip install -e /home/myproject/src
+  $ /home/myproject/venv/bin/pip install uwsgi
+
+
+.. note::
+
+  The exact commands for creating a virtual environment might differet based on
+  the Python version you are using and the system you are on. At the end of the day
+  we need a virtualenv in /home/myproject/venv with your project dependencies
+  installed.
+
+
+Okay, that is all for getting your application on the server and setting up the
+environment.
+
+Preparing your Application for Service
+''''''''''''''''''''''''''''''''''''''
+
+You likely have a way to configure your application, maybe with a ``create_api``
+function or something like that. This "thing" that configures your application
+gives you an instance of :any:`falcon.API`.
+
+We need to expose that in some way so that uWSGI can find it, we will do that by
+creating a wsgi.py file. The contents for your file should change bases on how
+your application is configured. When all is said and done, have a variable
+called ``application`` which is your :any:`falcon.API` instance.
+
+.. code-block:: python
+  :caption: /home/myproject/src/wsgi.py
+
+  import os
+  import myproject
+
+  config = myproject.get_config(os.environ['MYPROJECT_CONFIG'])
+  application = myproject.create_api(config)
+
+This is a bare bones wsgi file. It imports your application and creates the
+:any:`falcon.API` instance that represents our application, exposing it as the
+variable called ``application``. This file is what we will point uWSGI at to
+load your application.
+
+Note that we did not call ``run`` like a developer might do within a Flask
+application during development or use `wsgiref.simple_server.make_server` , we
+simply assigned our `API` instance to ``application``. We *do not* want to start
+a server, uWSGI will do that and starting one here will have unexpected results.
+
+
+Getting Falcon behind uWSGI
+'''''''''''''''''''''''''''
+
+With our wsgi.py file setting up our application it is time to configure uWSGI.
+To do this, we create a uwsgi.ini file. In general, you shouldn't commit this
+file, it's creation should be of your deployment tool which is targeting the
+environment which are deploying.
+
+This configuration, when executed, will create a new uWSGI server which will be
+pointed at our wsgi.py file and served on 127.0.0.1:8080.
+
+.. code-block:: ini
+  :caption: /home/myproject/src/uwsgi.ini
+
+  [uwsgi]
+  master = 1
+  vacuum = true
+  socket = 127.0.0.1:8080
+  enable-threads = true
+  thunder-lock = true
+  threads = 2
+  processes = 2
+  virtualenv = /home/myproject/venv
+  wsgi-file = /home/myproject/src/wsgi.py
+  chdir = /home/myproject/src
+  uid = myproject-runner
+  gid = myproject-runner
+
+
+.. note:: Thread vs Processes vs Both
```

Comment:
I would risk a guess that a very popular, if not the most popular, way to scale IO bound processes is using _gevent_ or similar concurrency libraries.
That is mentioned in several places in the Falcon documentation and the codebase [1], but I would think _gevent_ case also deserves a practical example how to configure stuff with uWSGI.

1. https://falcon.readthedocs.io/en/stable/user/faq.html#is-falcon-thread-safe